src/elements/content-sidebar/versions/VersionsSidebar.js

```diff
@@ -10,42 +10,71 @@ import InlineError from '../../../components/inline-error';
 import { LoadingIndicatorWrapper } from '../../../components/loading-indicator';
-import type { VersionActionCallback } from './Versions';
 import './VersionsSidebar.scss';
 
 type Props = {
     error?: string,
     fileId: string,
     isLoading: boolean,
-    onDelete: VersionActionCallback,
-    onDownload: VersionActionCallback,
-    onPreview: VersionActionCallback,
-    onPromote: VersionActionCallback,
-    onRestore: VersionActionCallback,
     parentName: string,
-    versions: Array<BoxItemVersion>,
+    versions?: Array<BoxItemVersion>,
 };
 
-const VersionsSidebar = ({ error, isLoading, parentName, ...rest }: Props) => (
-    <SidebarContent
-        className="bcs-Versions"
-        data-resin-component="preview"
-        data-resin-feature="versions"
-        title={
-            <React.Fragment>
-                <BackButton data-resin-target="back" to={`/${parentName}`} />
-                <FormattedMessage {...messages.versionsTitle} />
-            </React.Fragment>
-        }
-    >
-        <LoadingIndicatorWrapper className="bcs-Versions-content" crawlerPosition="top" isLoading={isLoading}>
-            {error && <InlineError title={<FormattedMessage {...messagesCommon.error} />}>{error}</InlineError>}
-            <VersionsList isLoading={isLoading} {...rest} />
-        </LoadingIndicatorWrapper>
-    </SidebarContent>
-);
+const VersionsSidebar = ({ error, isLoading, fileId, parentName, versions = [], ...rest }: Props) => {
+    const currentId = versions[0] ? versions[0].id : '';
+    const showVersions = !!versions.length;
+    const showEmpty = !isLoading && !showVersions;
+    const versionGroups = versions.reduce((groups, version) => {
+        const versionGroup = getGroup(version) || '';
+
+        groups[versionGroup] = groups[versionGroup] || [];
+        groups[versionGroup].push(version);
+
+        return groups;
+    }, {});
+
+    return (
+        <SidebarContent
+            className="bcs-Versions"
+            data-resin-component="preview"
+            data-resin-feature="versions"
+            title={
+                <React.Fragment>
+                    <BackButton data-resin-target="back" to={`/${parentName}`} />
+                    <FormattedMessage {...messages.versionsTitle} />
+                </React.Fragment>
+            }
+        >
+            <LoadingIndicatorWrapper className="bcs-Versions-content" crawlerPosition="top" isLoading={isLoading}>
+                {error && <InlineError title={<FormattedMessage {...messagesCommon.error} />}>{error}</InlineError>}
+
+                {showEmpty && (
+                    <div className="bcs-Versions-empty">
+                        <FormattedMessage {...messages.versionsEmpty} />
+                    </div>
+                )}
+
+                {showVersions && (
+                    <ul className="bcs-Versions-menu">
+                        {Object.keys(versionGroups).map(versionGroupKey => (
```

Comment:
Do we need to worry about sorting the order of the versionGroups? Assuming the API data comes back in a sorted order this should work, but I'm thinking about the corner case of if version has no `createdAt` (is that even possible?) in which case it gets no heading and has a key of `''` and I don't know where that would appear in the `Object.keys` order

Reply:
It doesn't look like `created_at` is a nullable field and the server explicitly sorts versions by that value for its response. I've removed the unnecessary existence checks from the new components.